app/build.gradle.kts

```diff
@@ -1,8 +1,63 @@
 @file:Suppress("UnstableApiUsage")
 
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+
+
 plugins {
+    alias(libs.plugins.kotlin.multiplatform)
+    alias(libs.plugins.kotlin.plugin.compose)
     alias(libs.plugins.android.application)
-    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.compose)
+}
+
+kotlin {
+    androidTarget {
+        compilations.all {
+            compileTaskProvider.configure {
+                compilerOptions {
+                    jvmTarget.set(JvmTarget.JVM_17)
+                }
+            }
+        }
+    }
+    jvm("desktop")
+
+    applyDefaultHierarchyTemplate()
+
+    sourceSets {
+        all {
+            languageSettings {
+                optIn("androidx.compose.ui.ExperimentalComposeUiApi")
+                optIn("com.popovanton0.blueprint.ExperimentalBlueprintApi")
+            }
+        }
+
+        commonMain.dependencies {
+            implementation(project(":blueprint"))
+            implementation(compose.foundation)
+            implementation(compose.material3)
+            implementation(compose.materialIconsExtended)
+            implementation(compose.preview)
+            implementation(compose.uiTooling)
+            implementation(compose.uiUtil)
+        }
+
+        androidMain.dependencies {
+            implementation(libs.androidx.coreKtx)
+            implementation(libs.androidx.lifecycleRuntimeKtx)
+            implementation(libs.androidx.activityCompose)
+            // note, these should have been debugImplementations,
+            // but that does not seem to work
+            implementation(libs.androidx.compose.ui.tooling)
+            implementation(libs.androidx.compose.ui.testManifest)
```

Comment:
I just tried adding this code to the end of the file, and it seems to work (android and desktop apps launch):

```kotlin
dependencies {
    debugImplementation(libs.androidx.compose.ui.tooling)
    debugImplementation(libs.androidx.compose.ui.testManifest)
}
```

Can you, please, also test this? Is it ok to do this in the KMP lib?

Reply:
Well, I'm no Gradle specialist (then again, who is?), but:

* this probably works because debug/release splits are only valid for Android libraries;
* so that's somewhat unintentional (the dependency is passed to Android targets because there's nowhere else for them to go);
* but that's not a "real" Gradle dependency block anyway (it's a `KotlinDependencyHandler` interface from kotlin gradle plugin).

I'm not sure whether using this unintentional method of getting previews to debug configurations only is going to break in future versions of Gradle and/or plugins